config_registry.go

```diff
@@ -0,0 +1,76 @@
+package evergreen
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/mongodb/grip"
+)
+
+var registry *configSectionRegistry
+
+func init() {
+	// add any new config sections to the variable below to register them
+	configSections := []configSection{
+		&AlertsConfig{},
+		&AmboyConfig{},
+		&APIConfig{},
+		&AuthConfig{},
+		&CloudProviders{},
+		&HostInitConfig{},
+		&JiraConfig{},
+		&LoggerConfig{},
+		&NewRelicConfig{},
+		&NotifyConfig{},
+		&RepoTrackerConfig{},
+		&SchedulerConfig{},
+		&ServiceFlags{},
+		&SlackConfig{},
+		&UIConfig{},
+	}
+
+	registry = newConfigSectionRegistry()
+	catcher := grip.NewSimpleCatcher()
+
+	for _, section := range configSections {
+		catcher.Add(registry.registerSection(section.id(), section))
+	}
+	if catcher.HasErrors() {
+		panic(fmt.Sprintf("error registering config sections: %s", catcher.String()))
+	}
```

Comment:
I might prefer having an init function in each of the file/files where a config block is defined that adds it to the registry, so that you can add a section editing only one or two files.

similarly, is it possible to write a test that will error if you add a section that isn't in the registry?

lastly, and along these lines, the registry is public, is there a way to move at least some of the code *out* of the top level package?

Reply:
Discussed in person, I'll make these changes after refactoring these into a different package (when we get rid of the file entirely). The tests will already fail if you add a section that isn't in the registry because GetConfig() will error